openpdf/src/main/java/com/lowagie/text/pdf/PdfWriter.java

```diff
@@ -448,7 +448,8 @@ int size() {
 
         void writeCrossReferenceTable(OutputStream os, PdfIndirectReference root, PdfIndirectReference info, PdfIndirectReference encryption, PdfObject fileID, int prevxref) throws IOException {
             int refNumber = 0;
-            if (writer.isFullCompression()) {
+            boolean useNewXrefFormat = writer.isFullCompression() || position > 9_999_999_999L;
+            if (useNewXrefFormat) {
```

Comment:
The old-style xref table only allots 10 digits for each object's offset. In this case, we must write an xref stream instead.

Reply:
would be nice to have this comment in the code itself for posterity!